modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/DefaultConfigurationExpirationMonitor.java

```diff
@@ -0,0 +1,201 @@
+        /**
+         * Validates the provided configuration and returns an actual {@link ConfigurationExpirationMonitor}.
+         *
+         * @return a {@link ConfigurationExpirationMonitor}
+         */
+        public ConfigurationExpirationMonitor build()
+        {
+            checkArgument(manager.extensionRegistry != null, "extensionRegistry cannot be null");
+            checkArgument(manager.muleContext != null, "muleContext cannot be null");
+            checkArgument(manager.frequency > 0, "frequency must be greater than zero");
+            checkArgument(manager.timeUnit != null, "timeUnit cannot be null");
+
+            return manager;
+        }
+
+    }
+
+    private ExtensionRegistry extensionRegistry;
+    private MuleContext muleContext;
+    private long frequency;
+    private TimeUnit timeUnit;
+    private BiConsumer<String, Object> expirationHandler;
+
+    private ScheduledExecutorService executor;
+
+    private DefaultConfigurationExpirationMonitor()
+    {
+    }
+
+    /**
+     * Starts a scheduler which fires expiration tasks on the given {@link #frequency} and executes
+     * the {@link #expirationHandler} on each matching configuration instance
+     *
+     * @throws MuleException if the manager could not be started
+     */
+    @Override
+    public void start() throws MuleException
+    {
+        executor = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, getPrefix(muleContext) + "extension.expiration.manager"));
+        executor.scheduleWithFixedDelay(() -> expire(), frequency, frequency, timeUnit);
+    }
+
+    private void expire()
+    {
+        if (muleContext.isStopping() || muleContext.isStopped())
+        {
+            return;
+        }
+
+        LOGGER.debug("Running configuration expiration cycle");
+        try
+        {
+            Map<String, Object> expired = extensionRegistry.getExpiredConfigInstances();
+            if (LOGGER.isDebugEnabled())
+            {
+                LOGGER.debug(expired.isEmpty()
+                             ? "No expired configuration instances were found"
+                             : "Found {} expired configurations", expired.size());
+            }
+
+            expired.entrySet().stream().forEach(this::handleExpiration);
+        }
+        catch (Exception e)
+        {
+            LOGGER.error("Found exception trying to expire idle configurations. Will try again on next cycle", e);
+        }
+
+    }
+
+    private void handleExpiration(Map.Entry<String, Object> config)
+    {
+        final String key = config.getKey();
+        LOGGER.debug("Expiring configuration of key {}", key);
+        try
+        {
+            expirationHandler.accept(key, config.getValue());
+            LOGGER.debug("Configuration of key {} was expired", key);
+        }
+        catch (Exception e)
+        {
+            LOGGER.error(String.format("Could not process expiration for dynamic config '%s' of type '%s'. Will try again on next cycle",
```

Comment:
no. As convenient as it is, the {} thing only works when you want to format a message without an exception. If you want the logger to also log the exception, then you need to build it yourself